### PR TITLE
Add a way to call for external moderator help

### PR DESCRIPTION
**Pull request by heinrich5991**
This is done by HTTP POSTing to a location specified by
`sv_modhelp_url`. We also provide a `src/modhelp/server.py` which can
use theses POSTs to forward them to Discord servers.

The POST contains a JSON object payload, with the keys `"port"` which
contains the server port, `"player_id"` which contains the calling
player's client ID, `"player_name"` which contains the calling player's
nick and `"message"` which is the user-specified message.

Make JSON-escaping function public, add tests and fix bugs uncovered by
these tests.

Supersedes #1129.

### CODE REVIEW

**Comment by heinrich5991** [comment]
@Learath2 @ChillerDragon What do you think?

File: src/game/server/ddracechat.cpp [hunk 1]
```diff
@@ -1396,23 +1397,29 @@ void CGameContext::ConModHelp(IConsole::IResult *pResult, void *pUserData)
...
+	if(!pPlayer)
 		return;
 
+	if(pPlayer->m_pPostJson)
+	{
+		pSelf->SendChatTarget(pResult->m_ClientID, "Your last request hasn't finished processing yet, please slow down");
+		return;
+	}
+
 	if(pPlayer->m_ModHelpTick > pSelf->Server()->Tick())
```
Comment (Learath2) [inline src/game/server/ddracechat.cpp]: Don't we usually store the last action and check if it + the delay is > Tick?

Comment (heinrich5991) [reply src/game/server/ddracechat.cpp]: Yes, we do, and also in this case. But I don't know what to do with the old request in that case.

**Comment by heinrich5991** [comment]
> Looks good, except maybe it shouldn't send the message along to discord if there already is a mod on the server that isn't afk.

Generally, we try to not leak whether there's a moderator on the server or not. Maybe I can include that in the POST request, so the HTTP endpoint can decide what to do.

File: src/game/server/ddracechat.cpp (hunk 1 above)
Comment (Learath2) [reply src/game/server/ddracechat.cpp]: Hmm, looked to me as `m_ModHelpTick` was only used for rate-limiting. Shouldn't matter anyway, just wanted to point it out :)

Comment (heinrich5991) [reply src/game/server/ddracechat.cpp]: Oh, I misunderstood.

Fixed.

**Pull request merged**